### PR TITLE
abuild.in: fix dbg function

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by maxice8** [comment]
@tcely

**Comment by tcely** [comment]
@maxice8 NACK. This doesn't fix the issue.

Here is the result of my debugging so far: http://tpaste.us/Lzx6

See https://github.com/alpinelinux/abuild/pull/71 for my fix.

Please close this PR.

File: abuild.in [hunk 1]
```diff
@@ -1685,17 +1685,16 @@ default_dbg() {
...
-		(
-			cd "$ddbg_srcdir"
-			XATTR=$(getfattr --match="" --dump "${ddbg_srcfile}")
-			${CROSS_COMPILE}objcopy --only-keep-debug "$ddbg_srcfile" "$ddbg_dstfile"
-			${CROSS_COMPILE}objcopy --add-gnu-debuglink="$ddbg_dstfile" "$ddbg_srcdir/$ddbg_srcfile"
-			mv "$ddbg_dstfile" "$ddbg_dstdir"
-			${CROSS_COMPILE}strip "$ddbg_srcfile"
-			[ -n "$XATTR" ] && { echo "$XATTR" | setfattr --restore=-; }
-		)
+		[ ! -d "$ddbg_dstdir" ] && mkdir -p "$ddbg_dstdir"
```
Comment (tcely) [inline abuild.in:1688]: I'm not sure why you changed the logic here, but the original makes more sense.

We could just always call `mkdir -p` as that won't fail if the directory exists or not.

File: abuild.in [hunk 2]
```diff
@@ -1685,17 +1685,16 @@ default_dbg() {
...
+
+		cd "$ddbg_srcdir"
+		XATTR=$(getfattr --match="" --dump "${ddbg_srcfile}")
+		${CROSS_COMPILE}objcopy --only-keep-debug "$ddbg_srcfile" "$ddbg_dstfile"
+		${CROSS_COMPILE}objcopy --add-gnu-debuglink="$ddbg_dstfile" "$ddbg_srcdir/$ddbg_srcfile"
+		mv "$ddbg_dstfile" "$ddbg_dstdir"
+		${CROSS_COMPILE}strip "$ddbg_srcfile"
+		[ -n "$XATTR" ] && { echo "$XATTR" | setfattr --restore=-; }
+		cd "$OLDPWD"
```
Comment (tcely) [inline abuild.in:1697]: As it turns out, adding any succeeding command after the test would work in the sub-shell.

**Pull request closed without merging**